controller/src/main/java/io/pravega/controller/server/eventProcessor/requesthandlers/StreamTask.java

```diff
@@ -28,4 +28,11 @@
      * @return future of processing
      */
     CompletableFuture<Void> writeBack(T event);
+
+    /**
+     * Method that indicates to the processor if it should ignore any fairness and start processing the task or not. 
+     * @param event event to process
+     * @return Completable Future which when completed will indicate if fairness should be ignore or honored. 
+     */
+    CompletableFuture<Boolean> ignoreFairness(T event);
```

Comment:
All implementations of `ignoreFairness` are simply comparing state, but I'm having a hard time convincing myself that this is sufficient to determine that it is the same task as before. Why can't it be a different task that leads to the same state?

Reply:
> Why can't it be a different task that leads to the same state?

Each type of task has a specific state associated with it. The metadata fence is taken via `state`. 
Also, there is exactly one task of a particular type that can be ongoing (this is determined via task specific record). For example, ongoing scale is identified by `EpochTransitionRecord`, ongoing update is determined by `ConfigurationRecord`, ongoing truncate is identified by `TruncationRecord` etc. 

So task processing happens as follows:
1. update task specific record to ensure only one outstanding task of a particular type exists. 
2. task specific event is posted and then is picked for processing
3. task processing checks if corresponding metadata record is set and then tries to take metadata fence via `state` 

So if `state` matches the state the task expects to set as a fence, then the task processing can run.